acceptable/mocks.py

```diff
@@ -276,7 +276,10 @@ class Endpoint(object):
     for this endpoint.
     """
     def __init__(self, base_url, service_name, endpoint_spec, response_callback=None):
-        self._url = urljoin(base_url, endpoint_spec.location)
+        if isinstance(endpoint_spec.location, str):
+            self._url = urljoin(base_url, endpoint_spec.location)
+        else:
+            self._url = endpoint_spec.location
```

Comment:
What's the expected type of `endpoint_spec.location` if not a string? A compiled regex, I guess from the PR title? In that case, is `base_url` just ignored? Sorry, this looks a bit cryptic to me; maybe I'm out of context. Would you please elaborate a bit on the use casec and maybe show a quick example? Thanks!

Reply:
The best example is code like below:

       EndpointSpec(
            'unscanned_download',
            re.compile(r'https?://\S+/unscanned-download/\S+'),
            ['GET'],
            None,
            None,
        ),

As there is currently no support for variables inside url's in Acceptable. The value of 'self.url' is directly passed to responses, so any invalid values will be handled there.

Reply:
The `EndpointSpec`s are automatically generated with service mocks from the API json definition, iiuc. And this url regex is passed to `responses.add_callback()` as you pointed, so that it'll be able to match test requests to multiple URLs. But, where does a developer using `acceptable` set this regex? Sorry, this is still cryptic to me; trying to understand the use case as an API developer. Thanks

Reply:
Unfortunately endpoints aren't generated for go services. And Acceptable doesn't seem to allow reading of openapi 3.0 formatted files. Which means that services have to be defined manually. Like the one above.

The service will be called by some url like: `http://localhost/unscanned-download/uuid_representing_the_download?token=abc` which, using responses on the back like `EndpointSpec` does, needs an url defined as a regexp.

Reply:
Oh I see, it makes sense now. Thanks for the explanation!